### PR TITLE
Fix Linux builds && Node version && add Windows support

### PR DESCRIPTION
**Pull request by pandasauce**
# Linux Builds

The CI build depends on the Microsoft Azure VM image for Ubuntu, which is far from a standard environment. ARM builds also require sudo and seem to assume that the build is being run as root in the first place.

This Dockerfile provides a fixed environment for building on Linux.

# Node 20

Building with Node 16 was failing with permission errors in `oclif`. Upgrading to Node 20 fixed this.

```
+ yarn package-deb
yarn run v1.22.22
$ oclif pack deb
 ›   Warning: oclif update available from 3.7.3 to 4.14.12.
oclif: gathering workspace for ns-cli to /build/cli/tmp/ns-cli
    Error: Command failed: npm pack --unsafe-perm
    /tmp/yarn--1722444948707-0.497377533169139/node: 3: exec: /root/.nvm/versions/node/v16.20.2/bin/node: Permission denied
    npm ERR! code 126
    npm ERR! path /build/cli
    npm ERR! command failed
    npm ERR! command sh -c -- yarn build && oclif manifest && oclif readme

    npm ERR! A complete log of this run can be found in:
    npm ERR!     /root/.npm/_logs/2024-07-31T16_55_49_476Z-debug-0.log

    Code: 126
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

# Windows Support

@1tacocat1's solution ran on Windows after fixing the builds, but existing path treatment was coded in a non-portable way, so it was failing to load the configuration. This is now fixed.

GitHub actions workflow was updated to include the dependencies for the Windows builds and to produce Windows artifacts.

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -483,7 +505,7 @@ DESCRIPTION
   Commands to retrieve assessment data
 ```
 
-_See code: [dist/commands/assessment/index.ts](https://github.com/cosdon/nowsecure-cli/blob/v1.0.0/dist/commands/assessment/index.ts)_
+_See code: [dist/commands/assessment/index.ts](https://github.com/cosdon/nowsecure-cli/blob/v1.1.0/dist/commands/assessment/index.ts)_
```
Comment (1tacocat1) [inline cli/README.md]: Update to https://github.com/nowsecure/nowsecure-cli/blob/v1.1.0/dist/commands/assessment/index.ts)_

File: cli/README.md [hunk 2]
```diff
@@ -813,7 +835,7 @@ FLAGS
   --ui=<value>           URL of the UI server
 ```
 
-_See code: [dist/commands/configure/index.ts](https://github.com/cosdon/nowsecure-cli/blob/v1.0.0/dist/commands/configure/index.ts)_
+_See code: [dist/commands/configure/index.ts](https://github.com/cosdon/nowsecure-cli/blob/v1.1.0/dist/commands/configure/index.ts)_
```
Comment (1tacocat1) [inline cli/README.md]: same as previous

File: cli/README.md [hunk 3]
```diff
@@ -1244,7 +1266,7 @@ DESCRIPTION
   Commands for users & accounts
 ```
 
-_See code: [dist/commands/user/index.ts](https://github.com/cosdon/nowsecure-cli/blob/v1.0.0/dist/commands/user/index.ts)_
+_See code: [dist/commands/user/index.ts](https://github.com/cosdon/nowsecure-cli/blob/v1.1.0/dist/commands/user/index.ts)_
```
Comment (1tacocat1) [inline cli/README.md]: same as previous

File: cli/package.json [hunk 4]
```diff
@@ -91,17 +91,18 @@
...
     "make-package": ".ci/package.sh",
     "test": "mocha --forbid-only \"test/**/*.test.ts\"",
     "version": "oclif readme && git add README.md",
     "link-deps": "yarn link @nowsecure/platform-lib @nowsecure/sarif @nowsecure/github-snapshot"
   },
   "engines": {
-    "node": ">=16.0.0"
+    "node": ">=20.0.0"
   },
   "bugs": "https://github.com/cosdon/nowsecure-cli/issues",
```
Comment (1tacocat1) [inline cli/package.json]: NowSecure...

**Review by 1tacocat1 — CHANGES_REQUESTED** [review]
Looks like some of the past work linked docs to @cosdon 's fork.  Should be updated.

**Comment by 1tacocat1** [comment]
looks good to me but @cosdon should look at to verify

**Review by cosdon — APPROVED** [review]
LGTM

**Pull request merged**